### PR TITLE
api of change password and change name

### PR DESCRIPTION
**Pull request by NasirAli1419**

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -98,6 +101,33 @@ exports.Register = (req, res) => {
...
+    let salt = bcrypt.genSaltSync(10);
+    let hash = bcrypt.hashSync(newPassword, salt); 
+    let passwordResult= await Users.findByIdAndUpdate({_id: userId},{Password: hash});
+    return response.Ok(passwordResult);
+}
+catch(e){
+   return response.BadRequest(e);
+}
+
+}
```
Comment (Sid12261996) [inline controllers/userController.js:128]: For changing password we should first check the older password match or not. Only If older password matches we will proceed to change the password.

File: models/userModel.js [hunk 2]
```diff
@@ -33,7 +35,9 @@ const User = new schema({
     // Post: ["Status", "Images", "Shared-post", "Videos"],
     JobProfile: [String],
     ImageUrl: {type: String, default: 'default'},
-    CoverUrl: String
+    CoverUrl: {type: schema.Types.String},
+    Status:{type: schema.Types.String}
```
Comment (Sid12261996) [inline models/userModel.js:39]: Why do we need this field here?? If yes, all those older users will have that field empty so need to be handling that as well...!

File: routes/userRoute.js [hunk 3]
```diff
@@ -9,6 +8,34 @@ router.post('/register',userController.Register);
 //User Login API endpoint
 router.post('/login',userController.Login);
 
+router.put('/changePassword', async (req, res) => {
```
Comment (Sid12261996) [inline routes/userRoute.js:11]: please add middleware JwtToken here, So only users who are authorized can access both the routes

File: routes/userRoute.js [hunk 4]
```diff
@@ -9,6 +8,34 @@ router.post('/register',userController.Register);
...
+catch(e){
+    console.log('Route is catching Error',e);
+    res.status(500).json(e);
+}
+
+});
+
+
+
+router.put('/nameUpdate', async (req, res) => {
```
Comment (Sid12261996) [inline routes/userRoute.js:27]: need JwtToken middleware before actually calling the controller

**Comment by Sid12261996** [comment]
please do resolve the comments.. IF any doubt pls connect with me

File: controllers/userController.js (hunk 1 above)
Comment (NasirAli1419) [reply controllers/userController.js:128]: Okay Sid.

**Pull request merged**